### ISSUE TITLE
Cube at wrong position in Chamber 13

### ISSUE THREAD

**Issue by Deconimus** [mwpenny/portal64-still-alive#11]
#### Version of the game
Master Branch, commit `d99af47`

#### Original Hardware or Emulator?
Project64 3.0.1

#### What is the bug?
The cube in the second room of Chamber 13 is not initially at the original location as in the original game. It should be on the inactive moving platform instead of on the ground.

#### Steps to reproduce the behavior:
1. In test chamber 13
2. Look through the glass.
3. See the cube in the second room being at the wrong position.

#### Screenshots
![GLideN64_Portal_64_001](https://github.com/mwpenny/portal64-still-alive/assets/2451901/7cadab3b-4296-4b70-ad9c-b995de46d776)
![GLideN64_Portal_64_000](https://github.com/mwpenny/portal64-still-alive/assets/2451901/02067b75-fd18-4665-b0ed-280387c913ad)

#### Additional context
The first screenshot shows the cube being at the wrong position as mentioned. The second was taken as I was trying out whether the current game physics are capable of handling a cube on a moving platform (which seems to work).

**Comment by mwpenny** [comment]
Thanks.

Here is how the cube is positioned in the level's `.blend` file (proper position):
![image](https://github.com/mwpenny/portal64-still-alive/assets/9356790/9002fe15-e45b-4de2-9a23-d734b0192464)

If I enter the level from the "new game" menu or from the previous chamber, it is placed correctly:
![portal_64-000](https://github.com/mwpenny/portal64-still-alive/assets/9356790/9bf2c9f9-1047-421c-98fe-d4fb24656665)

However, if I load a save, I can reproduce what you are seeing:
![portal_64-001](https://github.com/mwpenny/portal64-still-alive/assets/9356790/6367c44e-6b2d-4ddf-8048-864d2422336c).
I have debug contact solver drawing on, and can see the cube starting in its proper place before getting thrown across the room.

Likely, it is spawning inside of the platform and getting ejected. In fact, when I edit the level and move the cube slightly inside of the platform, I see the same thing. So it looks to be an issue with scene deserialization (affecting more than just cubes). This will be a low priority for now, but does need to be fixed.

**Comment by Deconimus** [comment]
You're right, I forgot to mention I was loading the level instead of starting a new game. Starting a new game at this chamber yields the cube being placed correctly as you've shown for me too.

**Comment by FoxbitDreamtail** [comment]
wouldn't spawning the cube slightly above the platform upon chamber load and then letting physics take the wheel fix this issue? it kind of looks to me like it is being flung off the platform on load, either that or their is a weird offset of some kind....

**Comment by mwpenny** [comment]
That would solve the problem in this one instance, but special casing things like this can get quite messy code-wise and also doesn't address the issue with different objects/platforms in other parts of the game. So a general solution is best.

This looks like an interaction between save file (de)serialization and rigid body sleep flags.
